pydevices/acq400Devices/acq2106_mgtdram8.py

```diff
@@ -0,0 +1,327 @@
+# modification, are permitted provided that the following conditions are met:
+#
+# Redistributions of source code must retain the above copyright notice, this
+# list of conditions and the following disclaimer.
+#
+# Redistributions in binary form must reproduce the above copyright notice, this
+# list of conditions and the following disclaimer in the documentation and/or
+# other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+import numpy as np
+import MDSplus
+import threading
+import Queue
+import socket
+import time
+import inspect
+
+
+try:
+    acq400_hapi = __import__('acq400_hapi', globals(), level=1)
+except:
+    acq400_hapi = __import__('acq400_hapi', globals())
+
+INPFMT = ':INPUT_%3.3d'
+
+class _ACQ2106_MGTDRAM8(MDSplus.Device):
+    """
+    D-Tacq ACQ2106_MGT support.
+
+    """
+
+    base_parts=[
+        # The user will need to change the hostname to the relevant hostname/IP.
+        {'path':':NODE', 'type':'text', 'value':'acq2106_999', 'options':('no_write_shot',)},
+        {'path':':CAP_BLOCKS', 'type':'numeric', 'value':40, 'options':('no_write_shot',)},
+        {'path':':SITE','type':'numeric', 'value': 1, 'options':('no_write_shot',)},
+        {'path':':TRIG_MODE','type':'text', 'value': 'role_default', 'options':('no_write_shot',)},
+        {'path':':ROLE','type':'text', 'value': 'master', 'options':('no_write_shot',)},
+        {'path':':FREQ','type':'numeric', 'value': int(1e7), 'options':('no_write_shot',)},
+        {'path':':HW_FILTER','type':'numeric', 'value':0, 'options':('no_write_shot',)},
+        {'path':':SEG_LENGTH','type':'numeric', 'value': 2*(2**20), 'options':('no_write_shot',)},
+        {'path':':MAX_SEGMENTS','type':'numeric', 'value': 2002, 'options':('no_write_shot',)},
+        {'path':':SEG_EVENT','type':'text', 'value': 'MGT_HOST_PULL', 'options':('no_write_shot',)},
+        {'path':':TRIG_TIME','type':'numeric', 'options':('write_shot',)},
+        {'path':':TRIG_STR','type':'text', 'options':('nowrite_shot',),'valueExpr':"EXT_FUNCTION(None,'ctime',head.TRIG_TIME)"},
+        {'path':':RUNNING','type':'any', 'options':('no_write_model',)},
+        ]
+    
+    debug=None
```

Comment:
this would override the Device.debug field that can be set via environment vars. check MDSplus/tree.py